### PR TITLE
Update setup documents

### PR DESCRIPTION
**Pull request by rantan**
I change the structure of setup.md to be placed as a part of tapyrus-core getting started document.
So I also updated tapyrus-core getting started document in a [PR](https://github.com/chaintope/tapyrus-core/pull/65). Check it please.

### CODE REVIEW

**Comment by Yamaguchi** [comment]
There are many typos or inconsistent uses of words in this pull request.

e.g.
- "Aggregate public key" or "aggregated public key"?
- Why the words "Aggregate public key" and "Node secret share" start with upper case ?

Could you fix them before reviewing?

Many tools will support you.

- https://www.deepl.com/
- https://chrome.google.com/webstore/detail/grammarly-for-chrome/kbfnbcaeplbcioakkpcpgfkobkghlhen

**Comment by rantan** [comment]
@Yamaguchi 

Thanks, I will check it again totally. but i will give you short reply for your comment.

> "Aggregate public key" or "aggregated public key"?

I used "Aggregate public key" because tapyrus-core uses this description in documents for consistency. (But i found a case i missed to change.)

> Why the words "Aggregate public key" and "Node secret share" start with upper case ?

I just imitated tapyrus-core document. It seems defined as Tapyrus network parameters. So, I should define Node secret share as a tapyrus-signer  parameter too if i use this kind of description.

**Comment by rantan** [comment]
@Yamaguchi I updated setup.md, federation.md and README.md. Can you check it again?

configuration.md is still old. I'm going to create another PR for updating it.

File: doc/setup.md [hunk 1]
```diff
@@ -15,13 +17,55 @@ We suppose (or recommend) that all signers can communicate with each other using
...
+Block VSS is a sort of VSS which is produced at 4.2 Signature Issuing Protocol in [the paper](https://doi.org/10.1007/3-540-47719-5_33).
+[The structure](#structure-of-vss) is mentioned later.
+All signers must generate Block VSSs and exchange each other. 
+
+**Block secret share**
+Block secret share is a share that is produced at 4.2 Signature Issuing Protocol in [the paper](https://doi.org/10.1007/3-540-47719-5_33).
+The value is 32bytes data of private key on secp256k1 curve.
+
+**Local signature**
+Local Signature is a signature, which is produced in each signer node at 4.2 Signature Issuing Protocol in [the paper](https://doi.org/10.1007/3-540-47719-5_33).
```
Comment (Yamaguchi) [inline doc/setup.md:46]: "Local Signature" or "Local signature"?

File: doc/setup.md [hunk 2]
```diff
@@ -141,41 +174,35 @@ output: {
 
 where:
 
-- `public_key[]` is an array of public keys generated by Signer[j] in Step 2.
-- `private_key[i]` is a private key of Signer[i] generated in Step 2.
+- `public_key[]` is an array of public keys generated by Signer[j].
+- `private_key[i]` is a private key of Signer[i] generated.
```
Comment (Yamaguchi) [inline doc/setup.md:178]: > - `private_key[i]` is a private key of Signer[i] generated.

I think the word "generated" is not required.

```
`private_key[i]` is a private key of Signer[i].
```

File: doc/setup.md [hunk 3]
```diff
@@ -15,13 +17,55 @@ We suppose (or recommend) that all signers can communicate with each other using
...
 
-## Step 2. Generate key pair
+**Node secret share**
+Node secret share is a share that is produced at 4.1 Key Generation Protocol in [the paper](https://doi.org/10.1007/3-540-47719-5_33).
+Each signer must have its Node secret share. 
+This is calculated by collected Node VSSs.
+The value is 32bytes data of private key on secp256k1 curve.
 
-Signer generates node key pairs using `tapyrus-setup createkey`.
+Tapyrus signer network uses the following unique parameters to each block generation round.
```
Comment (Yamaguchi) [inline doc/setup.md:34]: Should be "Tapyrus-signer network" or "Tapyrus Signer Network"

File: doc/setup.md [hunk 4]
```diff
@@ -41,9 +85,9 @@ And then, they sort received public keys.
...
 In the following steps, Signers are supposed to be sorted by public keys and indexed as Signer[i] (i = 1, 2, 3, ..., n)
 
-## Step 3. Generate node verifiable secret shares.
+### Step 2. Generate node verifiable secret shares.
 
-Signer[i] generates "node" VSS(verifiable secret share)[^4] with `tapyrus-setup createnodevss`.
+Signer[i] generates Node VSS with `tapyrus-setup createnodevss`.
 
 ```
 tapyrus-setup createnodevss --public_key=<public_key[1]> --public_key=<public_key[2]> ... --public_key=<public_key[n]> --private_key=<private_key[i]> --networkid=<networkid> --block_height=<block_height> --threshold=<t>
```
Comment (rantan) [inline doc/setup.md:93]: `--public_key` should be `--public-key`

Comment (rantan) [reply doc/setup.md:93]: `--networkid` and `block_height` shoud be removed here.

File: doc/setup.md [hunk 5]
```diff
@@ -57,34 +101,28 @@ output:
...
 where:
 
-- `public_key[]` is an array of public keys generated by Signer[j] in Step 2.
-- `private_key[i]` is a private key of Signer[i] generated in Step 2.
-- `networkid` is Network ID we choose in Step 1. we use it as a nonce of encryption.
-- `block_height` is a 64-bits integer to be applied to this node vss. In initial setup, this should be 0. This option is optional, default is 0.
+- `public_key[]` is an array of public keys generated by Signer[j] in Step 1.
+- `private_key[i]` is a private key of Signer[i] generated in Step 1.
+- `networkid` is Network ID we choose. We use it as a nonce of encryption.
+- `block_height` is a 64-bits integer to be applied to this federation. In the initial setup, this should be 0. This option is optional, and the default is 0.
```
Comment (rantan) [inline doc/setup.md:107]: `networkid` and `block_height` should be remove. it is removed from real command line arguments.

File: doc/setup.md [hunk 6]
```diff
@@ -57,34 +101,28 @@ output:
 
 where:
 
-- `public_key[]` is an array of public keys generated by Signer[j] in Step 2.
-- `private_key[i]` is a private key of Signer[i] generated in Step 2.
-- `networkid` is Network ID we choose in Step 1. we use it as a nonce of encryption.
-- `block_height` is a 64-bits integer to be applied to this node vss. In initial setup, this should be 0. This option is optional, default is 0.
+- `public_key[]` is an array of public keys generated by Signer[j] in Step 1.
+- `private_key[i]` is a private key of Signer[i] generated in Step 1.
```
Comment (rantan) [inline doc/setup.md:105]: Use kebab case.

File: doc/setup.md [hunk 7]
```diff
@@ -95,36 +133,31 @@ tapyrus-setup aggregate --vss=<node_vss[1, i]> --vss=<node_vss[2, i]> ... --vss=
 output: <aggregated_public_key> <node_secret_share[i]>
 ```
 
-- `node_vss[j, i]`(j = 1, 2, ..., n: i != j) are secret values produced in Step 3 and sent from other Signer[j].
-- `node_vss[i, i]` is produced by Signer[i] themselves in Step 3.
-- `private_key[i]` is the private key of Signer[i], generated in Step 2.
+- `node_vss[j, i]`(j = 1, 2, ..., n: i != j) are secret values produced in Step 2 and sent from other Signer[j].
+- `node_vss[i, i]` is produced by Signer[i] themselves in Step 2.
+- `private_key[i]` is the private key of Signer[i], generated in Step 1.
```
Comment (rantan) [inline doc/setup.md:138]: private key should use kebab case. It is also commandline agument.

File: doc/setup.md [hunk 8]
```diff
@@ -141,41 +174,35 @@ output: {
```
Comment (rantan) [inline doc/setup.md:174]: `--public_key` and `--private_key` should be kebab-case.

File: doc/setup.md [hunk 9]
```diff
@@ -141,41 +174,35 @@ output: {
...
-
-## Step 7. Sign the genesis block locally.
+### Step 2. Sign the genesis block locally.
 
-Signer[i] secretly receives `block_vss[j, i]` generated by other signers. Then Signer[i] generates `local_sig` using `tapyrus-setup sign`.
+Signer[i] secretly receives `block_vss[j, i]` generated by other signers. Then Signer[i] generates Local signature using `tapyrus-setup sign`.
 
 ```
 tapyrus-setup sign --block_vss=<block_vss[1, i]> --block_vss=<block_vss[2, i]> ... --block_vss=<block_vss[n, i]> --aggregated_public_key=<aggregated_public_key> --nodesecret=<node_secret_share[i]> --private_key=<private_key[i]> --block=<block>
```
Comment (rantan) [inline doc/setup.md:197]: `--threshold` also needs.

File: doc/federation.md [hunk 10]
```diff
@@ -4,38 +4,42 @@ This document describes the Federation Management of Tapyrus.
...
 
-The following describes how to change the federation.
+The following describes how to change the Federation.
 
-### Re-create aggregated public key and secret.
+### Re-create Aggregate public key and secret.
 
-Federation members, including the new signer, regenerate the aggregated public key and the secret in the same procedure as Step 2-Step 4 in [How to set up new Tapyrus Signer Network](./setup.md).
-Note that the same node public key can be used for the existing signer, so Step 2 is not required. However, for a new signer, a key pair needs to be created in Step 2.
+Federation members, including the new signer, regenerate the Aggregate public key and the secret in the same procedure as [Generate Aggregate public key and Node secret share for Tapyrus-signer network](./setup.md#generate-aggregate-public-key-and-node-secret-share-for-tapyrus-signer-network).
```
[the remark this answers is not in the document]
Comment (rantan) [reply doc/federation.md:24]: I think there is no reason. I will use Tapyrus Signer Network consistently.

File: doc/federation.md [hunk 11]
```diff
@@ -4,38 +4,42 @@ This document describes the Federation Management of Tapyrus.
...
+Note that the same node public key can be used for the existing signer, so Step 1 is not required. However, for a new signer, a key pair needs to be created in Step 1.
 
-### Updating the aggregated public key for existing signers
+### Updating the Aggregate public key for existing signers
 
-Update the aggregated public key held by the existing signer.
+:heavy_exclamation_mark:Caution: 
+> The RPC functionality is not implemented at 0.4.0 release. It is going to be implemented in a future release.
+> In 0.4.0 release, you can set federation information into `federations.toml` file directory. If you want to update federations settings, you can update the `federations.toml` and restart tapyrus-signerd.
```
[the remark this answers is not in the document]
Comment (rantan) [reply doc/federation.md]: > In 0.4.0 release, you can set federation information into federations.toml file directory.
>
>> maybe "federations.toml file" is OK. the word "directory" is not needed.

I missed taking the spell, "directly" is correct.

> If you want to update federations settings,
> 
>> you mean "federation's settings"?

Yes. I update it to just "settings".

File: doc/setup.md [hunk 12]
```diff
@@ -15,13 +17,55 @@ We suppose (or recommend) that all signers can communicate with each other using
...
 
-## Step 1. Choose Network ID
+## Tapyrus signer network parameters
+In addition to [Tapyrus network parameters](https://github.com/chaintope/tapyrus-core/blob/master/doc/tapyrus/getting_started.md#tapyrus-network-parameters), Tapyrus signer network uses following unique parameters to each tapyrus network.
 
-Choose Network ID, which is a 32-bit integer, and share among all signers.
+**Node VSS**  
+Node VSS is a sort of Verifiable Secret Share(VSS), which is produced at 4.1 Key Generation Protocol in [the paper](https://doi.org/10.1007/3-540-47719-5_33).
+[The structure](#structure-of-vss) is mentioned later.
+All signers must have Node VSSs from all signers. The VSSs must include its VSS  in federations.toml config file to startup a node.
```
[the remark this answers is not in the document]
Comment (rantan) [reply doc/setup.md]: It fixed.

File: doc/setup.md (hunk 3 above)
Comment (rantan) [reply doc/setup.md]: I fixed

File: doc/setup.md (hunk 1 above)
Comment (rantan) [reply doc/setup.md]: It is fixed.

File: doc/setup.md [hunk 13]
```diff
@@ -15,13 +17,55 @@ We suppose (or recommend) that all signers can communicate with each other using
...
+Local Signature is a signature, which is produced in each signer node at 4.2 Signature Issuing Protocol in [the paper](https://doi.org/10.1007/3-540-47719-5_33).
+By collecting Local signatures can generate a final signature, which can be verified by Aggregate public key, with calculating Lagrange interpolation.
+
+## Generate Aggregate public key and Node secret share for Tapyrus-signer network
+
+Here describes how each signer gets Aggregate public key of Tapyrus Signer Network(TSN) and Node secret share. 
+TSN produces signatures for blocks of the Tapyrus blockchain. 
+The signatures are equal with that signed by a private key, which is aggregated all signer's private key.
+So the signatures can be verified with the public key, which is aggregated all signer's public key. 
+However, the TNS doesn't use the aggregate private key to produce the signatures.
```
[the remark this answers is not in the document]
Comment (rantan) [reply doc/setup.md]: fixed

File: doc/setup.md (hunk 4 above)
Comment (rantan) [reply doc/setup.md]: fixed

File: doc/setup.md (hunk 6 above)
Comment (rantan) [reply doc/setup.md:108]: Here needs no change.

File: doc/setup.md (hunk 5 above)
Comment (rantan) [reply doc/setup.md]: removed

File: doc/setup.md [hunk 14]
```diff
@@ -57,34 +101,28 @@ output:
...
 
-[^4]: [Provably Secure Distributed Schnorr Signatures and a (t,n) Threshold Scheme for Implicit Cerfiticates](https://t.co/jMhQnovLcb)
-[^5]: [ChaCha20 and Poly1305 for IETF Protocols](https://tools.ietf.org/html/rfc8439)
+:heavy_exclamation_mark:Caution: 
+> The VSS encryption is not implemented at 0.4.0 release. It is going to be implemented in a future release.
 
-## Step 4. Generate an aggregated public key
+[^4]: [ChaCha20 and Poly1305 for IETF Protocols](https://tools.ietf.org/html/rfc8439)
+
+### Step 3. Generate an aggregated public key
```
[the remark this answers is not in the document]
Comment (rantan) [reply doc/setup.md]: fixed

File: doc/setup.md (hunk 7 above)
Comment (rantan) [reply doc/setup.md:141]: commandline argument is updated. here is no need to change.

File: doc/setup.md (hunk 8 above)
Comment (rantan) [reply doc/setup.md:179]: fixed

File: doc/setup.md (hunk 2 above)
Comment (rantan) [reply doc/setup.md]: fixed.

File: doc/setup.md (hunk 9 above)
Comment (rantan) [reply doc/setup.md]: fixed.

**Pull request merged**